azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java

```diff
@@ -937,8 +939,12 @@ public void accept(MessageSender a) {
         CompletableFuture<Map<String, Object>> rawdataFuture = new CompletableFuture<Map<String, Object>>();
         
         ManagementRetry retrier = new ManagementRetry(rawdataFuture, endTime, this.underlyingFactory, request);
-        Timer.schedule(retrier, Duration.ZERO, TimerType.OneTimeRun);
-        
+
+        final CompletableFuture<?> scheduledTask = this.timer.schedule(retrier, Duration.ZERO);
+        if (scheduledTask.isCompletedExceptionally()) {
+            ExceptionUtil.transferException(scheduledTask, rawdataFuture);
```

Comment:
Instead of explicitly completing an inner task, can this be handled inside scheduled task?